pandas/core/groupby/grouper.py

```diff
@@ -89,13 +114,85 @@ class Grouper:
     with a frequency of 60s
 
     >>> df.groupby(Grouper(level='date', freq='60s', axis=1))
+
+    If you want to adjust the start of the bins based on a fixed timestamp:
+
+    >>> start, end = "1/10/2000 02:00:00", "1/20/2000 02:00"
+    >>> rng = pd.date_range(start, end, freq="1231min")
+    >>> ts = pd.Series(np.arange(len(rng)) * 3, index=rng)
+    >>> ts
+    2000-01-10 02:00:00     0
+    2000-01-10 22:31:00     3
+    2000-01-11 19:02:00     6
+    2000-01-12 15:33:00     9
+    2000-01-13 12:04:00    12
+    2000-01-14 08:35:00    15
+    2000-01-15 05:06:00    18
+    2000-01-16 01:37:00    21
+    2000-01-16 22:08:00    24
+    2000-01-17 18:39:00    27
+    2000-01-18 15:10:00    30
+    2000-01-19 11:41:00    33
+    Freq: 1231T, dtype: int64
+    >>> ts.groupby(pd.Grouper(freq="2711min")).sum()
+    2000-01-10 00:00:00     9
+    2000-01-11 21:11:00    21
+    2000-01-13 18:22:00    33
+    2000-01-15 15:33:00    45
+    2000-01-17 12:44:00    57
+    2000-01-19 09:55:00    33
+    Freq: 2711T, dtype: int64
+    >>> ts.groupby(pd.Grouper(freq="2711min", origin=pd.Timestamp("1970-01-01"))).sum()
+    2000-01-08 11:44:00     0
+    2000-01-10 08:55:00     9
+    2000-01-12 06:06:00    21
+    2000-01-14 03:17:00    33
+    2000-01-16 00:28:00    72
+    2000-01-17 21:39:00    63
+    Freq: 2711T, dtype: int64
+
+    If you want to adjust the start of the bins with an offset, the two following
+    lines are equivalent:
+
+    >>> ts.groupby(pd.Grouper(freq="2711min", origin="1/10/2000 02:00:00")).sum()
+    2000-01-10 02:00:00     9
+    2000-01-11 23:11:00    21
+    2000-01-13 20:22:00    33
+    2000-01-15 17:33:00    45
+    2000-01-17 14:44:00    90
+    Freq: 2711T, dtype: int64
+    >>> ts.groupby(pd.Grouper(freq="2711min", offset="2h")).sum()
+    2000-01-10 02:00:00     9
+    2000-01-11 23:11:00    21
+    2000-01-13 20:22:00    33
+    2000-01-15 17:33:00    45
+    2000-01-17 14:44:00    90
+    Freq: 2711T, dtype: int64
+
+    To replace the use of the deprecated `base` argument:
+    >>> # ts.groupby(pd.Grouper(freq="2711min", base=2)).sum()
+    >>> #   becomes:
+    >>> ts.groupby(pd.Grouper(freq="2711min", offset="2min")).sum()
+    2000-01-10 00:02:00     9
+    2000-01-11 21:13:00    21
+    2000-01-13 18:24:00    33
+    2000-01-15 15:35:00    45
+    2000-01-17 12:46:00    57
+    2000-01-19 09:57:00    33
+    Freq: 2711T, dtype: int64
     """
 
     _attributes: Tuple[str, ...] = ("key", "level", "freq", "axis", "sort")
 
     def __new__(cls, *args, **kwargs):
         if kwargs.get("freq") is not None:
             from pandas.core.resample import TimeGrouper
+            from pandas.core.resample import _validate_resample_deprecated_args
+
```

Comment:
Well. I wanted to avoid a circular imports here. Do you have a better solution ?
The real issue is that the warning needs to appear properly between resample AND pd.Grouper...

I explained the full issue (and an alternative solution) in an earlier comment of mine here:
https://github.com/pandas-dev/pandas/pull/31809#discussion_r379860446

So. What

Reply:
umm, don't validate here at all. you can simply validate all fo the args IN TimeGrouper itself. doing it here and there is so confusing.